### DISCUSSION TITLE
Creating Nodes

### DISCUSSION THREAD

**Discussion by BrechtCorbeel** [Comfy-Org/ComfyUI#3746] [General]
Ok, so I have a few years of python experience, I know python is loaded into memory, I also know work-arounds that would allow me to update code with new code, but to implement it might be a security risk to do so, but given restarting comfyui takes about 3 minutes for me how do you guys test nodes and implementations?

I might want to push stuff that I created myself as a contributor, but I guess if I have to put in hacks that circumvent certain issues like needing to restart comfy, which would need to be pulled out at some point, is there a visual studio code bridge or how do you guys do it?

Or would a bridge or ability to implement code on the fly be a good idea as in a sort of developer mode that you can turn on or off I am guessing this part is nothing new, but I wonder about best practices/common practices for coding in ComfyUI?

**Comment by ltdrdata** [comment]
I'm not sure about your environment, but when I test my nodes, I do it in an environment where the installation of custom nodes is minimized. It takes less than 10 seconds to restart.

**Reply by BrechtCorbeel** [reply]
How much of a risk do you think an eval() node would be?

If I put up a github repo with my custom nodes how or what do I add to make sure they are properly installed or stored or does it automatically put it in the right folder, etc. and work like that if it works as a custom node on my system in general?

**Reply by ltdrdata** [reply]
Using `eval` is not a good idea.
For pip packages, you should use `requirements.txt`. Additionally, `install.py` is supported for other methods, but I am considering changing this approach in the long term.
